subsys/net/lib/nrf_cloud/src/nrf_cloud_codec.c

```diff
@@ -980,11 +981,48 @@ int nrf_cloud_obj_location_request_create(struct nrf_cloud_obj *const obj,
 		goto cleanup;
 	}
 
-	/* By default, nRF Cloud will send the location to the device */
-	if (!request_loc &&
-	    nrf_cloud_obj_num_add(&data_obj, NRF_CLOUD_LOCATION_KEY_DOREPLY, 0, false)) {
+	if (config && (obj->type == NRF_CLOUD_OBJ_TYPE_JSON)) {
+		/* Assume no differences from defaults. */
 		err = -ENOMEM;
-		goto cleanup;
+		if (config->do_reply != true) {
+			err = nrf_cloud_obj_bool_add(&config_obj,
+						     NRF_CLOUD_LOCATION_JSON_KEY_DOREPLY,
+						     config->do_reply, false);
+			if (err) {
+				goto cleanup;
+			}
+		}
+		if (config->hi_conf != false) {
+			err = nrf_cloud_obj_bool_add(&config_obj,
+						     NRF_CLOUD_LOCATION_JSON_KEY_HICONF,
+						     config->hi_conf, false);
+			if (err) {
+				goto cleanup;
+			}
+		}
+		if (config->fallback != true) {
+			err = nrf_cloud_obj_bool_add(&config_obj,
+						     NRF_CLOUD_LOCATION_JSON_KEY_FALLBACK,
+						     config->fallback, false);
+			if (err) {
+				goto cleanup;
+			}
+		}
+		if (!err) {
+			/* At least one entry differed from defaults, so add it to the obj. */
+			err = nrf_cloud_obj_object_add(obj, NRF_CLOUD_LOCATION_JSON_KEY_CONFIG,
+						       &config_obj, false);
+			if (err) {
+				goto cleanup;
+			}
+			/* The config object now belongs to the location request object */
+			config_obj.json = NULL;
```

Comment:
this line can be removed, `nrf_cloud_obj_object_add` resets it.

```
/**
 * @brief Add a key string and object to the provided object.
 *
 * @details If successful, obj_to_add will be reset with @ref nrf_cloud_obj_reset
 *          since its data has been moved to obj.
```

Reply:
it looks like there is an errant blank line in the header comments for `nrf_cloud_obj_object_add`... could you please remove it 😄